code/dal.py

```diff
@@ -1,11 +1,15 @@
 import MySQLdb
 import numpy as np
 import os
+import random
 import string
 import time
+import utils
 
+from nltk.tokenize.stanford import StanfordTokenizer
 
-def get_emails(num_emails=100, fetch_all=False):
+
+def get_emails(num_emails=100, max_users=150, fetch_all=False):
```

Comment:
I don't think its worth having num_emails and max_users together. We should keep only max_users
If someone asks num_emails to be 5k, then this code will select the smallest file (e.g 5k_emails.csv or 10k_emails.csv) that satisfies the requirement effectively negating the work of max_users

Or even better, we can have a separate get_emails(max_users) that always reads from the entire dataset and filters based on the number of users required

Reply:
Plus it will be much simpler and more intuitive if we return the caller exactly the number of users requested

Reply:
Yes, these are conflicting requirements.

Going forward, should we retain the num_emails option (two functions)? 
Or is it enough to just have max_users?